src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs

```diff
@@ -2272,12 +2272,23 @@ public bool TryDequeueWaiter([MaybeNullWhen(false)] out TaskCompletionSourceWith
                 return false;
             }
 
-            public bool TryPeekRequest([MaybeNullWhen(false)] out HttpRequestMessage request)
+            public bool TryPeekUncancelledRequest(HttpConnectionPool pool, [MaybeNullWhen(false)] out HttpRequestMessage request)
             {
-                if (_queue is not null && _queue.TryPeek(out QueueItem item))
+                if (_queue is not null)
                 {
-                    request = item.Request;
-                    return true;
+                    while (_queue.TryPeek(out QueueItem item))
+                    {
+                        if (item.Waiter.Task.IsCanceled)
+                        {
+                            if (NetEventSource.Log.IsEnabled()) pool.Trace("Discarding canceled request from queue.");
```

Comment:
Log the request as well?

Reply:
Looks like we don't include it in other traces in `HttpConnectionPool`.

E.g. compared to
https://github.com/dotnet/runtime/blob/96ef47b2c6d6591cebba4eb260763576621e8236/src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpConnectionPool.cs#L1792

I'll leave it off of this PR, but we could look at improving these logs overall to reference the request where possible.